### PR TITLE
Do not build mina on single-node-tests job

### PR DESCRIPTION
**Pull request by dkijania**
Explain your changes:
Skip building mina on single node tests. Instead, use already built mina lightnet packages. Command line tests evolved from being a test with dev profile (kind of unit test but on already built mina ) to became component test for release version of mina. Execution time is similar (~14 min) but we are testing on higher level and more production alike mina.

Explain how you tested your changes:
Run CI and observe green single node tests

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

File: buildkite/scripts/single-node-tests.sh [hunk 1]
```diff
@@ -2,27 +2,23 @@
...
 
-eval "$(opam config env)"
-export PATH="/home/opam/.cargo/bin:/usr/lib/go/bin:$PATH"
-export GO=/usr/lib/go/bin/go
+git config --global --add safe.directory /workdir
 
-# TODO: Stop building lib_p2p multiple times by pulling from buildkite-agent artifacts or docker or somewhere
-echo "--- Build libp2p_helper TODO: use the previously uploaded build artifact"
-make -C src/app/libp2p_helper
+source buildkite/scripts/export-git-env-vars.sh
```
Comment (SanabriaRusso) [inline buildkite/scripts/single-node-tests.sh:13]: Just a though: I guess this won't work for forks until you merge #14596

File: buildkite/src/Jobs/Test/SingleNodeTest.dhall [hunk 2]
```diff
@@ -26,18 +26,25 @@ let JobSpec = ../../Pipeline/JobSpec.dhall
...
+let dependsOn = DebianVersions.dependsOn DebianVersions.DebVersion.Bullseye Profiles.Type.Standard
+
+
+let buildTestCmd : Size -> Command.Type = \(cmd_target : Size) ->
+  let key = "single-node-tests" in
   Command.build
     Command.Config::{
-      commands = RunInToolchain.runInToolchain ["DUNE_INSTRUMENT_WITH=bisect_ppx", "COVERALLS_TOKEN"] "buildkite/scripts/single-node-tests.sh ${path} && buildkite/scripts/upload-partial-coverage-data.sh ${key} dev",
-      label = "${profile} single-node-tests",
+      commands = RunInToolchain.runInToolchain ["DUNE_INSTRUMENT_WITH=bisect_ppx", "COVERALLS_TOKEN"] "buildkite/scripts/single-node-tests.sh && buildkite/scripts/upload-partial-coverage-data.sh ${key}",
```
Comment (SanabriaRusso) [inline buildkite/src/Jobs/Test/SingleNodeTest.dhall:42]: I understand that no passing a $PROFILE to `upload-partial-coverage-data.sh` then prevents this pipeline build from building the package with `opam`.

**Comment by SanabriaRusso** [comment]
I was not of much help regarding OCaml, but overall pipeline flow was clear. Liked the idea to keep pushing logic to scripts and removing them from Dhall expressions. Makes pipelines clearer.

**Review by SanabriaRusso — COMMENTED** [review]
I think we might need extra eyes for the OCaml part.

File: buildkite/scripts/single-node-tests.sh (hunk 1 above)
Comment (dkijania) [reply buildkite/scripts/single-node-tests.sh:13]: yes, However we are planning to merge a couple commits from rampup to  berkeley soon

File: buildkite/src/Jobs/Test/SingleNodeTest.dhall (hunk 2 above)
Comment (dkijania) [reply buildkite/src/Jobs/Test/SingleNodeTest.dhall:42]: yes. If dev parameter is provided then  we need  run bisect reporter through opam

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Pull request merged**